### ISSUE TITLE
Update release process to use tags

### ISSUE THREAD

**Issue by breedx-splk** [open-telemetry/opentelemetry-android#262]
The current release process performs a release from a branch, typically `main`. This is risky, because there is a window of time during which unwanted changes could be merged. 

I think we should create and push a tag _first_, and the release build can then target that tag. That should give us significantly more safety around locking in what we intend to publish.

**Comment by trask** [comment]
another option is to copy-paste(-ish) the release infra from the other java repos and use release branches

**Comment by breedx-splk** [comment]
> another option is to copy-paste(-ish) the release infra from the other java repos and use release branches

Yeah true. I don't know those in enough detail, but I think having the consistency could help quite a bit.
